src/libs/antares/study/parts/short-term-storage/container.cpp

```diff
@@ -91,24 +92,44 @@ bool STStorageInput::saveToFolder(const std::string& folder) const
 {
     // create empty list.ini if there's no sts in this area
     Yuni::IO::Directory::Create(folder);
-    Yuni::IO::File::CreateEmptyFile(folder + SEP + "list.ini");
-    logs.notice() << "created empty ini: " << folder + SEP + "list.ini";
+    const std::string pathIni(folder + SEP + "list.ini");
+    IniFile ini;
 
-    return std::all_of(storagesByIndex.cbegin(), storagesByIndex.cend(), [&folder](auto& storage) {
-        return storage->saveProperties(folder);
+    logs.debug() << "saving file " << pathIni;
+
+    std::for_each(storagesByIndex.cbegin(), storagesByIndex.cend(), [&ini](auto& storage) {
+        return storage.saveProperties(ini);
     });
+
+    return ini.save(pathIni);
 }
 
 bool STStorageInput::saveDataSeriesToFolder(const std::string& folder) const
 {
     Yuni::IO::Directory::Create(folder);
     return std::all_of(storagesByIndex.cbegin(), storagesByIndex.cend(), [&folder](auto& storage) {
-        return storage->saveSeries(folder + SEP + storage->id);
+        return storage.saveSeries(folder + SEP + storage.id);
     });
 }
 
 std::size_t STStorageInput::count() const
 {
-    return storagesByIndex.size();
+  return std::count_if(storagesByIndex.begin(),
+                       storagesByIndex.end(),
+                       [](const STStorageCluster& st) {
+                           return st.properties.enabled;
+                       });
 }
+
+uint STStorageInput::removeDisabledClusters()
+{
+    const auto& it = std::remove_if(storagesByIndex.begin(), storagesByIndex.end(),
+        [](const auto& c) { return c.enabled(); });
```

Comment:
```suggestion
    const auto& it = std::remove_if(storagesByIndex.begin(), storagesByIndex.end(),
        std::mem_fn(&STStorageCluster::enabled));
```
C++20 makes this so much simpler...